components/elm/cime_config/config_component.xml

```diff
@@ -217,34 +217,35 @@
-    <desc compset="_ELM%[^_]*CNRDCTCBC"  >alm1.0 C-N, nutirent competition via relative demand, ctc soil cascade with black carbon deposition:</desc>
-    <desc compset="_ELM%[^_]*CNPRDCTCBC" >alm1.0 C-N-P, nutirent competition via relative demand, ctc soil cascade with black carbon deposition:</desc>
-    <desc compset="_ELM%[^_]*CNECACTCBC" >alm1.0 C-N, nutirent competition via equilibrium chemistry approximation, ctc soil cascade with black carbon deposition:</desc>
-    <desc compset="_ELM%[^_]*CNPECACTCBC">alm1.0 C-N-P, nutirent competition via equilibrium chemistry approximation, ctc soil cascade with black carbon deposition:</desc>
-    <desc compset="_ELM50"               >clm5.0 physics:</desc>
-    <desc compset="_ELM50%[^_]*SP"       >clm5.0 Satellite phenology:</desc>
-    <desc compset="_ELM50%[^_]*BGC"      >clm5.0 bgc (cn and methane):</desc>
+    <desc compset="_ELM"               >elm physics:</desc>
+    <desc compset="_ELM%[^_]*SP"       >elm Satellite phenology:</desc>
+    <desc compset="_ELM%[^_]*SPBC"     >elm Satellite phenology with black carbon deposition:</desc>
+    <desc compset="_ELM%[^_]*CN"       >elm cn:</desc>
+    <desc compset="_ELM%[^_]*CNDV"     >elm cn with dynamic vegetation:</desc>
+    <desc compset="_ELM%[^_]*BGC"      >elm bgc (cn and methane):</desc>
+    <desc compset="_ELM%[^_]*CROP"     >elm prognostic crop:</desc>
+    <desc compset="_ELM%[^_]*VIC"      >elm vic hydrology:</desc>
+    <desc compset="_ELM%[^_]*FATES"    >elm FATES (Functionally Assembled Terr. Ecosystem Simulator): (experimental)</desc>
+    <desc compset="_ELM%[^_]*BGCDV"    >elm BGC (CN with vertically resolved soil BGC, based on Century with Methane) with dynamic veg</desc>
+    <desc compset="_ELM%[^_]*CRDCTCBC"   >elm C only, nutirent competition via relative demand, ctc soil cascade with black carbon deposition:</desc>
+    <desc compset="_ELM%[^_]*CNRDCTCBC"  >elm C-N, nutirent competition via relative demand, ctc soil cascade with black carbon deposition:</desc>
+    <desc compset="_ELM%[^_]*CNPRDCTCBC" >elm C-N-P, nutirent competition via relative demand, ctc soil cascade with black carbon deposition:</desc>
+    <desc compset="_ELM%[^_]*CNECACTCBC" >elm C-N, nutirent competition via equilibrium chemistry approximation, ctc soil cascade with black carbon deposition:</desc>
+    <desc compset="_ELM%[^_]*CNPECACTCBC">elm C-N-P, nutirent competition via equilibrium chemistry approximation, ctc soil cascade with black carbon deposition:</desc>
+    <desc compset="_ELM50"               >elm physics:</desc>
+    <desc compset="_ELM50%[^_]*SP"       >elm Satellite phenology:</desc>
+    <desc compset="_ELM50%[^_]*BGC"      >elm bgc (cn and methane):</desc>
+    <desc compset="_ELM%[^_]*CNPRDCTCBCTOP" >elm C-N-P, nutrient competition via relative demand, ctc soil cascade with black carbon deposition and subgrid solar radiation parameterization :</desc>
```

Comment:
@wlin7 do you know how to use regex to distinguish between "CNPRDCTCBC" and "CNPRDCTBCTOP" ?

Reply:
Need an anchor  "$"